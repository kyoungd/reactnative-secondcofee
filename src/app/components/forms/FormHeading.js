import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'

const Title = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 20;
`

const Description = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  margin-top: 4;
`

export default function FormHeading({ title, description }) {
  return (
    <>
      {!!title && <Title numberOfLines={1}>{title}</Title>}
      <Description numberOfLines={2}>{description}</Description>
    </>
  )
}

FormHeading.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
}
