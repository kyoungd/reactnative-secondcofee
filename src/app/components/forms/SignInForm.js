import React from 'react'
import PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import styled from 'styled-components/native'
import { Formik } from 'formik'
import * as yup from 'yup'
import { FormField, RegularButton, CustomActivityIndicator } from '../ui'

const Wrapper = styled.View`
  padding: 16px;
`

const LogoWrapper = styled.View`
  height: 80;
  justify-content: center;
  align-items: center;
`

const Logo = styled.Image.attrs({
  resizeMode: 'contain',
  source: require('assets/images/logo.png')
})`
  width: 60;
  height: 60;
`

const ResetPasswordText = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  text-align: center;
`

const initialValues = {
  email: '',
  password: ''
}

const validationSchema = yup.object().shape({
  email: yup
    .string()
    .required('email address is required')
    .email('enter a valid email address'),
  password: yup.string().required('password is required')
})

export default function SignInForm({
  loading,
  onSubmit,
  onResetPasswordPress
}) {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={values => onSubmit(values)}
      render={props => (
        <KeyboardAwareScrollView enableOnAndroid>
          <Wrapper>
            <LogoWrapper>
              <Logo />
            </LogoWrapper>

            <FormField
              label="Email"
              keyboardType="email-address"
              value={props.values.email}
              onChangeText={text => props.setFieldValue('email', text)}
              error={props.touched.email && props.errors.email}
            />

            <FormField
              label="Password"
              secure
              value={props.values.password}
              onChangeText={text => props.setFieldValue('password', text)}
              error={props.touched.password && props.errors.password}
            />

            <RegularButton
              text="SIGN IN"
              onPress={() => props.handleSubmit()}
            />

            <ResetPasswordText onPress={() => onResetPasswordPress()}>
              Forgot password?
            </ResetPasswordText>

            {loading && <CustomActivityIndicator />}
          </Wrapper>
        </KeyboardAwareScrollView>
      )}
    />
  )
}

SignInForm.propTypes = {
  loading: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onResetPasswordPress: PropTypes.func.isRequired
}
