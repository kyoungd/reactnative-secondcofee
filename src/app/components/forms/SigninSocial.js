import React from 'react'
import PropTypes from 'prop-types'
import View from 'react-native'
import styled from 'styled-components/native'
import { Formik } from 'formik'
import * as yup from 'yup'
import { FormField, RegularButton, CustomActivityIndicator } from '../ui'

const Wrapper = styled.View`
  padding: 16px;
`

const LogoWrapper = styled.View`
  height: 80;
  justify-content: center;
  align-items: center;
`

const Logo = styled.Image.attrs({
  resizeMode: 'contain',
  source: require('assets/images/logo.png')
})`
  width: 60;
  height: 60;
`

const ResetPasswordText = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  text-align: center;
`

export default function SignInSocial({
  loading,
  onSubmit,
  onResetPasswordPress
}) {
  return (
   <View>
    <Wrapper>

      <RegularButton
        text="Facebook"
        onPress={() => props.handleFacebook()}
      />

      <RegularButton
        text="Google"
        onPress={() => props.handleGoogle()}
      />

      <RegularButton
        text="Twitter"
        onPress={() => props.handleTwitter()}
      />

    </Wrapper>
  </View>
  )
}

SignInSocial.propTypes = {
  handleFacebook: PropTypes.func.isRequired,
  handleGoogle: PropTypes.func.isRequired,
  handleTwitter: PropTypes.func.isRequired
}