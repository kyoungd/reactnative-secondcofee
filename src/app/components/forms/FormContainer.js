import React from 'react'
import PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import styled from 'styled-components/native'
import { RegularButton, CustomActivityIndicator } from '../ui'
import FormHeading from './FormHeading'

const Wrapper = styled.View`
  padding: 16px;
`

export default function FormContainer({
  headingTitle,
  headingDescription,
  submitButtonText,
  onSubmitButtonPress,
  loading,
  children
}) {
  return (
    <KeyboardAwareScrollView enableOnAndroid>
      <Wrapper>
        <FormHeading title={headingTitle} description={headingDescription} />

        {children}

        {!loading && (
          <RegularButton
            text={submitButtonText}
            onPress={() => onSubmitButtonPress()}
          />
        )}
        {loading && <CustomActivityIndicator />}
      </Wrapper>
    </KeyboardAwareScrollView>
  )
}

FormContainer.propTypes = {
  headingTitle: PropTypes.string,
  headingDescription: PropTypes.string.isRequired,
  submitButtonText: PropTypes.string.isRequired,
  onSubmitButtonPress: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
}
FormContainer.defaultProps = {
  headingTitle: ''
}
