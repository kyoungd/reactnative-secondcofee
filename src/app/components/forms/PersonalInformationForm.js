import React from 'react'
import PropTypes from 'prop-types'
import { Formik } from 'formik'
import * as yup from 'yup'
import { FormField } from '../ui'
import FormContainer from './FormContainer'

export const initialValues = {
  fullName: 'John Appleseed',
  email: 'john@appleseed.com'
}

export const validationSchema = yup.object().shape({
  fullName: yup.string().required('full name is required'),
  email: yup
    .string()
    .required('email address is required')
    .email('enter a valid email address')
})

export default function PersonalInformationForm({ onSubmit }) {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={values => onSubmit(values)}
      render={props => (
        <FormContainer
          headingTitle="Personal Information"
          headingDescription="Tell us a little about you"
          submitButtonText="NEXT"
          onSubmitButtonPress={() => props.handleSubmit()}
          loading={false}
        >
          <FormField
            label="Full Name"
            value={props.values.fullName}
            onChangeText={text => props.setFieldValue('fullName', text)}
            error={props.touched.fullName && props.errors.fullName}
          />

          <FormField
            disabled
            label="Email Address"
            keyboardType="email-address"
            value={props.values.email}
            onChangeText={text => props.setFieldValue('email', text)}
            error={props.touched.email && props.errors.email}
          />
        </FormContainer>
      )}
    />
  )
}

PersonalInformationForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
}
