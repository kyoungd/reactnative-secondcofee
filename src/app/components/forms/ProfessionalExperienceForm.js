import React from 'react'
import PropTypes from 'prop-types'
import { Formik } from 'formik'
import * as yup from 'yup'
import { FormField } from '../ui'
import FormContainer from './FormContainer'

const initialValues = {
  qualification: '',
  field: '',
  experience: ''
}

const validationSchema = yup.object().shape({
  qualification: yup.string().required('qualification is required'),
  field: yup.string().required('field is required'),
  experience: yup.string().required('years of experience is required')
})

export default function ProfessionalExperienceForm({ onSubmit }) {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={values => onSubmit(values)}
      render={props => (
        <FormContainer
          headingTitle="Professional Experience"
          headingDescription="Provide some details about your career"
          submitButtonText="NEXT"
          onSubmitButtonPress={() => props.handleSubmit()}
          loading={false}
        >
          <FormField
            label="Qualification"
            value={props.values.qualification}
            onChangeText={text => props.setFieldValue('qualification', text)}
            error={props.touched.qualification && props.errors.qualification}
          />

          <FormField
            label="Field/Category"
            value={props.values.field}
            onChangeText={text => props.setFieldValue('field', text)}
            error={props.touched.field && props.errors.field}
          />

          <FormField
            keyboardType="number-pad"
            label="Years of Experience"
            value={props.values.experience}
            onChangeText={text => props.setFieldValue('experience', text)}
            error={props.touched.experience && props.errors.experience}
          />
        </FormContainer>
      )}
    />
  )
}

ProfessionalExperienceForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
}
