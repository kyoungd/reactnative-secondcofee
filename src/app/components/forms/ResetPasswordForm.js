import React from 'react'
import PropTypes from 'prop-types'
import { Formik } from 'formik'
import * as yup from 'yup'
import { FormField } from '../ui'
import FormContainer from './FormContainer'

export const initialValues = {
  email: ''
}

export const validationSchema = yup.object().shape({
  email: yup
    .string()
    .required('email address is required')
    .email('enter a valid email address')
})

export default function ResetPasswordForm({ loading, onSubmit }) {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={values => onSubmit(values)}
      render={props => (
        <FormContainer
          headingDescription="Enter your email address and we'll help reset your password"
          submitButtonText="CONTINUE"
          onSubmitButtonPress={() => props.handleSubmit()}
          loading={loading}
        >
          <FormField
            label="Email Address"
            keyboardType="email-address"
            value={props.values.email}
            onChangeText={text => props.setFieldValue('email', text)}
            error={props.touched.email && props.errors.email}
          />
        </FormContainer>
      )}
    />
  )
}

ResetPasswordForm.propTypes = {
  loading: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired
}
