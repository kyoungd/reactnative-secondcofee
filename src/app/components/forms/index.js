import FormHeading from './FormHeading'
import SignInForm from './SignInForm'
import SignUpForm from './SignUpForm'
import ResetPasswordForm from './ResetPasswordForm'
import PersonalInformationForm from './PersonalInformationForm'
import ProfessionalExperienceForm from './ProfessionalExperienceForm'

export {
  FormHeading,
  SignInForm,
  SignUpForm,
  ResetPasswordForm,
  PersonalInformationForm,
  ProfessionalExperienceForm
}
