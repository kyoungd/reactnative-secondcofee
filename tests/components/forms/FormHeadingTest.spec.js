import React from 'react'
import { shallow } from 'react-native-testing-library'
import { FormHeading } from 'components/forms'

test('Component renders correctly', () => {
  const { output } = shallow(
    <FormHeading
      title="A form heading"
      description="Some text that supports the heading"
    />
  )
  expect(output).toMatchSnapshot()
})
