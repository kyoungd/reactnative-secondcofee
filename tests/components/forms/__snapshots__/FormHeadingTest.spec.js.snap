// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<React.Fragment>
  <ForwardRef
    numberOfLines={1}
  >
    A form heading
  </ForwardRef>
  <ForwardRef
    numberOfLines={2}
  >
    Some text that supports the heading
  </ForwardRef>
</React.Fragment>
`;
